notebooks/working/investigating_deconfounder.py

```diff
@@ -0,0 +1,592 @@
+
+from IPython import display
+import seaborn as sns
+
+import tensorflow as tf
+import numpy as np
+import numpy.random as npr
+import pandas as pd
+import tensorflow as tf
+import tensorflow_probability as tfp
+import statsmodels.api as sm
+
+from tensorflow_probability import edward2 as ed
+from sklearn.datasets import load_breast_cancer
+from pandas.plotting import scatter_matrix
+from scipy import sparse, stats
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LogisticRegression 
+from sklearn.metrics import confusion_matrix, classification_report, roc_auc_score, roc_curve
+
+import matplotlib
+matplotlib.rcParams.update({'font.sans-serif' : 'Helvetica',
+                            'axes.labelsize': 10,
+                            'xtick.labelsize' : 6,
+                            'ytick.labelsize' : 6,
+                            'axes.titlesize' : 10})
+import matplotlib.pyplot as plt
+
+import seaborn as sns
+color_names = ["windows blue",
+               "amber",
+               "crimson",
+               "faded green",
+               "dusty purple",
+               "greyish"]
+colors = sns.xkcd_palette(color_names)
+sns.set(style="white", palette=sns.xkcd_palette(color_names), color_codes = False)
+
+
+import os
+os.listdir('.')
+
+# +
+drive_alone_graph = CausalGraphicalModel(
+    nodes=['a', 'b', 'c', 'confounder',
+          'd',  'e', 'f', 'y'],
+    edges=[
+        ("a", "y"), 
+        ("b", "y"), 
+        ("c", "y"), 
+        ("d", "y"), 
+        ("e", "y"), 
+        ("f", "y"), 
+        ("confounder", "y"), 
+        
+        
+        ("confounder", "a"),
+        ("confounder", "b"),
+#         ("confounder", "c"),
+        ("confounder", "d"),
+        ("confounder", "e"),
+        ("confounder", "f"),
+
+
+        
+        
+        ("confounder", "y"), 
+        ("confounder", "y"), 
+    ]
+)
+
+# draw return a graphviz `dot` object, which jupyter can render
+drive_alone_graph.draw()
+
+# +
+sample_size = 10000
+
+confounder = np.random.normal(loc=20, scale = 5, size = sample_size)
+
+# +
+a = .3*confounder + np.random.normal(loc=10,scale=3,size=sample_size)
+b = -1*confounder + np.random.normal(loc=6,scale=3,size=sample_size)
+c = np.random.normal(loc=15,scale=3,size=sample_size) # + .5*confounder 
+d = .8*confounder + np.random.normal(loc=-10,scale=3,size=sample_size)
+e = .5*confounder + np.random.normal(loc=8,scale=3,size=sample_size)
+f= -.3*confounder + np.random.normal(loc=-12,scale=3,size=sample_size)
+y = 5 + 2*a + 2*b - 5*c + 3*d + 2*e - 5*f + 7*confounder + np.random.normal(loc=0,scale=1,size=sample_size)
+
+coeffs_true = [5,2,2,-5,3,2,-5,7]
+
+# plt.hist(confounder, bins = 50)
+
+# plt.hist(f, bins = 50)
+# -
+
+df = pd.DataFrame(np.array([a,b,c,d,e,f,confounder, y]).T, columns = ['a', 'b', 'c',
+          'd',  'e', 'f', 'confounder', 'y'])
+df.head()
+
+# +
+# sns.pairplot(df, size=1.5)
+
+# +
+ols_formula = "y ~ a+b+c+d+e+f"#+confounder "
```

Comment:
`model_full` should be full outcome model for `y` right?
```suggestion
ols_formula = "y ~ a+b+c+d+e+f+confounder "
```